Library/Homebrew/cask/audit.rb

```diff
@@ -270,6 +273,83 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def extract_url(url)
+      url.to_s.gsub(%r{.*"([a-z0-9]+://[^"]+)".*}m, '\1')
+    end
+
+    def remove_non_ascii(string)
+      string.gsub(/\P{ASCII}/, "")
+    end
+
+    def homepage
+      URI(remove_non_ascii(extract_url(cask.homepage))).host
+    end
+
+    def domain
+      strip_url_scheme(extract_url(cask.url)).gsub(%r{^([^/]+).*}, '\1')
```

Comment:
I think this can just be `URI(cask.url.to_s).host`.

Reply:
Sure